documentation/modules/con-securing-kafka-authentication.adoc

```diff
@@ -0,0 +1,115 @@
+//
+// assembly-securing-kafka-brokers.adoc
+
+[id='con-securing-kafka-authentication-{context}']
+= Listener authentication
+
+You can create `plain` (without encryption) or `tls` listener types.
+For clients outside the Kubernetes cluster, you create `external` listeners and specify a connection mechanism, which can be `nodeport`, `loadbalancer`, `ingress` or `route` (on OpenShift).
+
+Supported authentication options:
+
+. Mutual TLS authentication (only on the listeners with TLS encryption)
+. SCRAM-SHA authentication
+. xref:assembly-oauth-authentication_str[OAuth 2.0 token based authentication]
+
+.Kafka listener authentication options
+image::listener-config-options.png[options for listener authentication configuration]
+
+The listener `authentication` property is used to specify an authentication mechanism specific to that listener.
+
+The listener authentication mechanism is defined by the `type` field.
+If no `authentication` property is specified then the listener does not authenticate clients which connect through that listener.
+The listener will accept all connections without authentication.
+
+Authentication must be configured when using the User Operator to manage `KafkaUsers`.
+
+The following example shows:
+
+* A `plain` listener configured for SCRAM-SHA authentication
+* A `tls` listener with mutual TLS authentication
+* An `external` listener with mutual TLS authentication
+
+.An example showing listener authentication configuration
+[source,yaml,subs="attributes+"]
+----
+# ...
+listeners:
+  plain:
+    authentication:
+      type: scram-sha-512
+  tls:
+    authentication:
+      type: tls
+  external:
+    type: loadbalancer
+    tls: true
+    authentication:
+      type: tls
+# ...
+----
+
+[id='con-mutual-tls-authentication-{context}']
+== Mutual TLS authentication
+
+Mutual TLS authentication is always used for the communication between Kafka brokers and ZooKeeper pods.
+
+For mutual authentication or two-way authentication, both the server and the client present certificates.
+Strimzi can configure Kafka to use TLS (Transport Layer Security) to provide encrypted communication between Kafka brokers and clients either with or without mutual authentication. When you configure mutual authentication, the broker authenticates the client and the client authenticates the broker.
+
+NOTE: TLS authentication is more commonly one-way, with one party authenticating the identity of another. For example, when HTTPS is used between a web browser and a web server, the server obtains proof of the identity of the browser.
+
+.When to use mutual TLS authentication for clients
+
+Mutual TLS authentication is recommended for authenticating Kafka clients when:
+
+* The client supports authentication using mutual TLS authentication
+* It is necessary to use the TLS certificates rather than passwords
+* You can reconfigure and restart client applications periodically so that they do not use expired certificates.
```

Comment:
To be honest, I have some doubts about these: 
* This `The client supports authentication using mutual TLS authentication` is a prerequisite but we do not recommend to use TLS client authentication just because the client supports it. 
* The second point `It is necessary to use the TLS certificates rather than passwords` is very unspecific. It would be good to explain why would it be necessary.
* The third point `You can reconfigure and restart client applications periodically so that they do not use expired certificates.` is imho confusing / misleading since the periodically could be for example every 5 years.

Maybe we should just remove it if we do not have anything better? (I don't right now)

Reply:
Okay. Let's remove these '_when to use..._' sections. I've added a sentence to the top of the section:

> The authentication option you choose depends on how you wish to authenticate client access to Kafka brokers.